torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1090,6 +1090,25 @@ def generator():
     return list(generator())
 
 
+def sample_inputs_nn_activation_softmax(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        ((), (0, )),
+        ((S, ), (0, )),
+        ((S, S), (0, )),
+        ((S, S), (1, )),
+        ((S, M, S), (2, ))
+    )
+
+    def generator():
+        for shape, dim in cases:
+            yield SampleInput(make_arg(shape), args=(dim))
+            yield SampleInput(make_arg(shape), args=(dim), kwargs={'dtype': dtype})
```

Comment:
Passing the dtype kwarg as the same dtype of the tensor probably doesn't do much.

One option would be to detect if the requested tensor dtype is float32 or float16, for example, and then pass a dtype here of "double."